datafusion/sql/src/expr/identifier.rs

```diff
@@ -135,16 +138,17 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
                     let nested_name = nested_names[0].to_string();
 
                     let col = Expr::Column(Column::from((qualifier, field)));
-                    if let Some(udf) =
-                        self.context_provider.get_function_meta("get_field")
```

Comment:
I wonder could we plan the whole compound identifier, the benefit of `plan_*` function is able to customize the operator (in this case is compound identifier i.e. `a.b.c`) and expressions we get.

Reply:
Ideally i wanted to do that same, but this function uses `IdentNormalizer` field of `SqlToRel` struct. I am just thinking a way around & wanted to discuss the same here.

Reply:
We can use the result of `self.normalizer.normalize()` as the planner arguments, similar to the spirit of others function that we take the result of `sql_expr_to_logical_expr` as the planner arguments

Reply:
How about this revision ? Do you think we aligned here ? There is still more work to do.

Reply:
`fn sql_compound_identifier_to_expr` does many things, handles the logic for variable, compund indentifier & compund indentifier in outer query. In this change we are only supporting compund indentifier & not adding support compund indentifier in outer query.